iroha/benches/sumeragi.rs

```diff
@@ -13,7 +13,8 @@ fn get_n_peers(n: usize) -> Vec<PeerId> {
 }
 
 fn sort_peers(criterion: &mut Criterion) {
-    let mut network_topology = NetworkTopology::new(&get_n_peers(N_PEERS), None, 1);
+    let mut network_topology = NetworkTopology::new(&get_n_peers(N_PEERS), None, 1)
+        .expect("Failed to initialize topology.");
```

Comment:
It's better to separate constructor `new` and initialization `init`, why so - constructor means constructing object and usually it should not fail in runtime - because all parameters are statically typed and can be checked at compile time.
So like we did with `Kura` - we may introduce new `init` method.
If you want to prevent usage of not initialized `NetworkTopology` you may introduce two types - `NetworkTopology` with only `new` method and `InitializedNetworkTopology` with all other methods.

Reply:
I suggest to do it in a separate PR. https://jira.hyperledger.org/secure/RapidBoard.jspa?rapidView=264&projectKey=IR&view=detail&selectedIssue=IR-766